src/navigation/menu_bar.py

```diff
@@ -89,14 +109,32 @@ def do_save_as_file(self) -> None:
         """Try to Save As a file under a new name/path."""
 
         async def coroutine() -> None:
+            """
+            Prompt the user for a file path to save their note.
+
+            If the path entered is a valid file name, save the current note at that path.
+            """
             open_dialog = TextInputDialog(
                 title="Save As", label_text="Enter the path of the file:"
             )
-
-            path = await self.show_dialog_as_float(open_dialog)
-            self.application_state.current_path = path
-            if path := self.application_state.current_path:
+            user_entered_path = await self.show_dialog_as_float(open_dialog)
+            # Validate that the user entered path is
+            # 1. Not an empty string or None
+            # 2. Doesn't consist exclusively of whitespace
+            # 3. Isn't the string "." or ".."
+            # 4. if the file already exists warning of over writing
+            if (
+                user_entered_path
+                and not str.isspace(user_entered_path)
+                and user_entered_path not in [".", ".."]
+                and os.path.isfile(user_entered_path)
+            ):
+                path = NOTES_DIR + "/" + user_entered_path
```

Comment:
Would it be better to use `os.path.join` here? I think forward slashes are okay here, even if the user's machine is Windows.